src/axolotl/integrations/kernels/libs/dsv4/gated_pool.py

```
@@ -0,0 +1,90 @@
+    dout = tl.load(DOUT + m * D + d, mask=dmask, other=0.0).to(tl.float32)[None, :]  # [1, BD]
+    dkv = p * dout
+    dgate = p * dout * (kv - out)
+    tl.store(DKV + off, dkv, mask=cmask)
+    tl.store(DGATE + off, dgate, mask=cmask)
+
+
+class _GatedPool(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, kv, gate):
+        M, W, D = kv.shape
+        kv = kv.contiguous(); gate = gate.contiguous()
```

Ruff (0.15.17)

[error] Multiple statements on one line (semicolon)

(E702)